Modified benchmark-mbedtls script to print a comparison table

sergioVaz00 opened the pull request:
```
Step               	Without Codee	With Codee
=====================	=============	==========
CI: SS: Clone        	27.518 s		27.518 s
CI: BS: pwdirectives 	--- s		0.189 s
CI: BS: make all     	10.466 s		10.655 s
CI: BS: Test aes/cmac	0.005 s		0.005 s
CI: DS: Benchmark    	6.015 s		6.012 s
```

sergioVaz00 committed:
Modified benchmark-mbedtls script to print a comparison table

ulisescosti commented on benchmark-mbedTLS-vector.sh:
```diff
@@ -29,18 +30,23 @@ printf "##################################################\n"
 printf "Wellcome to Codee's interactive demo with MBedTLS\n"
 printf "##################################################\n"
 printf "Seven steps:\n"
```
```suggestion
printf "Five steps:\n"
```

ulisescosti commented on benchmark-mbedTLS-vector.sh:
```diff
@@ -171,35 +170,56 @@ printf "cmac algorithm\n"
+printf "5/5. Verifying the speedup ...\n"
 printf "##################################################\n"
 read -p "Press enter to continue"
 printf "\n"
```
```suggestion
printf "\n"
printf "Running MbedTLS benchmarks (with the original code). . .\n\n"
```

ulisescosti commented on benchmark-mbedTLS-vector.sh:
```diff
@@ -171,35 +170,56 @@ printf "cmac algorithm\n"
 printf "cmac algorithm\n"
 printf "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 build/programs/test/benchmark aes_cmac
 printf "original done\n"
```
```suggestion
printf "Original benchmarks were successfully executed \n\n"

printf "##################################################\n\n"
printf "Running MbedTLS benchmarks (with the vectorized code). . .\n\n"
```

ulisescosti commented on benchmark-mbedTLS-vector.sh:
```diff
@@ -171,35 +170,56 @@ printf "cmac algorithm\n"
+printf "cmac algorithm\n"
+printf "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
 buildVec/programs/test/benchmark aes_cmac
 printf "vectorized done\n"
```
```suggestion
printf "The vectorized benchmarks were successfully executed \n\n"
```

ulisescosti commented on benchmark-mbedTLS-vector.sh:
```diff
@@ -171,35 +170,56 @@ printf "cmac algorithm\n"
 #===============================================================================
 
-printf "\nDone.\n"
+tSource=$(bc -l <<< "($tSource1 - $tSource0) /1000")
+tBuildO=$(bc -l <<< "($tBuild1 - $tBuild0) /1000")
+tCodee=$(bc -l <<< "($tBuild3 - $tBuild2) /1000")
+tBuildC=$(bc -l <<< "($tBuild5 - $tBuild4) /1000")
+tTestO=$(bc -l <<< "($tBuild7 - $tBuild6) /1000")
+tTestC=$(bc -l <<< "($tBuild8 - $tBuild7) /1000")
+tDeployO=$(bc -l <<< "($tDeploy1 - $tDeploy0) /1000")
+tDeployC=$(bc -l <<< "($tDeploy2 - $tDeploy1) /1000")
+
+printf "\nStep               \tWithout Codee\tWith Codee\n"
+printf "=====================\t=============\t==========\n"
+printf "CI: SS: Clone        \t%.3f s\t\t%.3f s\n" $tSource $tSource
+printf "CI: BS: pwdirectives \t--- s\t\t%.3f s\n" $tCodee
+printf "CI: BS: make all     \t%.3f s\t\t%.3f s\n" $tBuildO $tBuildC
+printf "CI: BS: Test aes/cmac\t%.3f s\t\t%.3f s\n" $tTestO $tTestC
+printf "CI: DS: Benchmark    \t%.3f s\t\t%.3f s\n" $tDeployO $tDeployC
```
I think it is useless to compare the execution time here, since these MbedTLS benchmarks are not benchmarking time, but throughput instead.
These tests are calculating the amount of operations that are performed per unit of time. In this case, if I am not wrong, the benchmarks are implemented with a fixed time and a variable input workload.
Comparing raw times would be correct in exec. time benchmarks, where the time is variable, and the input workload is fixed.
We need to compare throughput metrics, not raw time numbers. For example, the average throughput, the average cycles/byte, etc.

ulisescosti reviewed (commented):
The reduction of the number of steps is OK (I just left a few minor suggestions).
But I think that we definetly need to change the metrics used to generate the last summary table. We can not compare raw time from benchmarks that are not testing execution times.

sergioVaz00 committed:
MbedTLS-benchmark: fix script to allow warnings

The CMAKE flag "MBEDTLS_FATAL_WARNINGS=Off" was not used for the first build.
This was making the build to fail with some compilers.
GCC 8.2 (oficially supported by MbedTLS) was detecting a warning in the
first build, and so, it was unable to finish the build.

sergioVaz00 committed:
Modified benchmark-mbedtls script to print a Speedup table

sergioVaz00 committed:
Improved and updated benchmark-mbedTLS-vector.sh with news algorithms
